### PR TITLE
Added rsvp endpoint for events, handles nonaffiliate rsvp for now.

### PR DESCRIPTION
**Pull request by thaigillespie**

### CODE REVIEW

File: src/services/EventService.ts [hunk 1]
```diff
@@ -71,4 +75,11 @@ export class EventService {
 
     return newAttendance;
   }
+
+  async registerForEventRSVP(eventId: number, appUser: AppUser): Promise<RSVP> {
```
Comment (godwinpang) [inline src/services/EventService.ts:79]: drop the for

Comment (godwinpang) [reply src/services/EventService.ts:79]: can do so for event attendance as well

Comment (godwinpang) [reply src/services/EventService.ts:79]: also missing jsdoc?

File: src/services/RSVPService.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+@singleton()
+export class RSVPService {
+  private rsvpRepository: Repository<RSVP>;
+
+  constructor(@inject(RSVPRepositoryToken) rsvpRepository: Repository<RSVP>) {
+    this.rsvpRepository = rsvpRepository;
+  }
+
+  async registerRSVP(event: Event, appUser: AppUser): Promise<RSVP | undefined> {
```
Comment (godwinpang) [inline src/services/RSVPService.ts:15]: jsdoc

File: src/controllers/EventController.ts [hunk 3]
```diff
@@ -2,16 +2,17 @@ import { JsonController, Param, Get, Post, Delete, Body, OnUndefined } from 'rou
 import { singleton, inject } from 'tsyringe';
 import { ResponseSchema } from 'routing-controllers-openapi';
```
Comment (godwinpang) [inline src/controllers/EventController.ts:4]: see previous comments

File: src/mappers/RSVPMapper.ts [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+import { RSVPService } from '@Services';
+import { RSVPRepositoryToken } from '@Repositories';
+
+import { Repository } from 'typeorm';
+import { classToPlain, plainToClass } from 'class-transformer';
+import { singleton, inject } from 'tsyringe';
+
+@singleton()
+export class RSVPMapper {
+  private rsvpRepository: Repository<RSVP>;
```
[the remark this answers is not in the document]
Comment (thaigillespie) [reply src/mappers/RSVPMapper.ts:12]: So I just remove the whole constructor then?

[the remark this answers is not in the document]
Comment (thaigillespie) [reply src/mappers/RSVPMapper.ts:12]: @godwinpang

[the remark this answers is not in the document]
Comment (godwinpang) [reply src/mappers/RSVPMapper.ts:12]: yeah it should default to empty constructor which is fine

[the remark this answers is not in the document]
Comment (thaigillespie) [reply src/mappers/RSVPMapper.ts:12]: Sounds good. Ty for letting me know

**Pull request merged**